### ISSUE TITLE
create_compute_pipeline does not accept enums.AutoLayoutMode for layout arg

### ISSUE THREAD

**Issue by dskkato** [pygfx/wgpu-py#422]
ooking at the interface of create_compute_pipeline, it seems that enums.AutoLayoutMode can be specified for the layout parameter.

```py
    def create_compute_pipeline(
        self,
        *,
        label="",
        layout: "Union[GPUPipelineLayout, enums.AutoLayoutMode]",
        compute: "structs.ProgrammableStage",
    ):
```

However, when actually specifying this option, the following error occurred:

```sh
(.venv) [23:59:41] dskkato:wgpu-py $ python repro.py
No config found!
EGL says it can present to the window but not natively
Traceback (most recent call last):
  File "repro.py", line 23, in <module>
    compute_pipeline = device.create_compute_pipeline(
  File "/home/dskkato/workspace/wgpu-py/wgpu-py/wgpu/backends/wgpu_native/_api.py", line 1174, in create_compute_pipeline
    layout=layout._internal,
AttributeError: 'str' object has no attribute '_internal'
```

The code I tried is as follows:

```py
import wgpu
import wgpu.utils

device = wgpu.utils.get_default_device()

shader_source = """

@group(0) @binding(0)
var<storage,read> data1: array<i32>;

@group(0) @binding(1)
var<storage,read_write> data2: array<i32>;

@compute
@workgroup_size(1)
fn main(@builtin(global_invocation_id) index: vec3<u32>) {
    let i: u32 = index.x;
    data2[i] = data1[i];
}
"""
cshader = device.create_shader_module(code=shader_source)

compute_pipeline = device.create_compute_pipeline(
    label="", layout=wgpu.enums.AutoLayoutMode.auto,
    compute={"module": cshader, "entry_point": "main"},
)
```

To fix this, it might be a good idea to check if auto is specified for layout and, if so, feed ffi.NULL to the layout of WGPUComputePipelineDescriptor struct.

**Comment by dskkato** [comment]
I'm not entirely sure what state the layout of the GPUComputePipeline should be in after specifying "auto", so it doesn't seem as straightforward as just accepting "auto".

**Comment by dskkato** [comment]
In the WebGPU specification, the details about the "auto" behavior are explained in this section:

https://www.w3.org/TR/webgpu/#shader-module-compilation-hints

Additionally, information about the default pipeline layout can be found here:

https://www.w3.org/TR/webgpu/#abstract-opdef-default-pipeline-layout

Therefore, it would be beneficial to follow the "auto" layout mode as described in these references if possible.

**Comment by dskkato** [comment]
closed by #423
